cpp/cmake/thirdparty/get_raft.cmake

```diff
@@ -14,12 +14,23 @@
 # limitations under the License.
 #=============================================================================
 
+set(CUML_MIN_VERSION_raft "${CUML_VERSION_MAJOR}.${CUML_VERSION_MINOR}.00")
+set(CUML_BRANCH_VERSION_raft "${CUML_VERSION_MAJOR}.${CUML_VERSION_MINOR}")
+
 function(find_and_configure_raft)
 
-    set(oneValueArgs VERSION FORK PINNED_TAG USE_RAFT_NN USE_FAISS_STATIC)
+    set(oneValueArgs VERSION FORK PINNED_TAG USE_RAFT_NN USE_FAISS_STATIC CLONE_ON_PIN)
     cmake_parse_arguments(PKG "${options}" "${oneValueArgs}"
             "${multiValueArgs}" ${ARGN} )
 
+    if(PKG_CLONE_ON_PIN AND NOT PKG_PINNED_TAG STREQUAL "branch-${CUML_BRANCH_VERSION_raft}")
+        message("Pinned tag found: ${PKG_PINNED_TAG}. Cloning raft locally.")
+        execute_process(
+                COMMAND git clone "https://github.com/${PKG_FORK}/raft.git" --branch ${PKG_PINNED_TAG} raft-src
+                WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/_deps)
+        set(CPM_raft_SOURCE ${CMAKE_CURRENT_BINARY_DIR}/_deps/raft-src)
+    endif()
```

Comment:
We should be able to do this in `rapids_cpm_find` with  using the new `CPM_<PACKAGE>_DOWNLOAD` added by @robertmaynard if I'm not mistaken, otherwise we should also be able to do this with `CPM_DOWNLOAD_ALL`

Reply:
When I tried using `CPM_DOWNLOAD_ALL`, it downloaded ALL of the dependencies and thus caused cuml to build everything (including FAISS). I didn't see a good way to turn that off. I didn't see another way to do this, but let me try `CPM_<package>_DOWNLOAD` to see if it works.

Reply:
We can use `CPM_<PackageName>_DOWNLOAD` once https://github.com/rapidsai/rapids-cmake/pull/158/ has been merged.